components/service/experiments/src/test/java/mozilla/components/service/experiments/ExperimentsUpdaterTest.kt

```diff
@@ -0,0 +1,138 @@
+    private lateinit var experimentSource: KintoExperimentSource
+    private lateinit var experimentsUpdater: ExperimentsUpdater
+    private lateinit var workRequest: PeriodicWorkRequest
+
+    private fun getDefaultExperimentStorageMock(): FlatFileExperimentStorage {
+        val storage: FlatFileExperimentStorage = mock()
+        `when`(storage.retrieve()).thenReturn(ExperimentsSnapshot(emptyList(), null))
+        return storage
+    }
+
+    private fun getDefaultExperimentSourceMock(): KintoExperimentSource {
+        val source: KintoExperimentSource = mock()
+        `when`(source.getExperiments(mozilla.components.support.test.any())).thenReturn(ExperimentsSnapshot(emptyList(), null))
+        return source
+    }
+
+    private fun resetExperiments(
+        source: KintoExperimentSource = getDefaultExperimentSourceMock(),
+        storage: FlatFileExperimentStorage = getDefaultExperimentStorageMock(),
+        valuesProvider: ValuesProvider = ValuesProvider()
+    ) {
+        // Initialize WorkManager (early) for instrumentation tests.
+        WorkManagerTestInitHelper.initializeTestWorkManager(context)
+
+        configuration = Configuration()
+        experiments = spy(ExperimentsInternalAPI())
+        experiments.valuesProvider = valuesProvider
+
+        `when`(glean.isInitialized()).thenReturn(true)
+        `when`(experiments.getGlean()).thenReturn(glean)
+        experimentStorage = storage
+        `when`(experiments.getExperimentsStorage(context)).thenReturn(storage)
+
+        experimentsUpdater = spy(ExperimentsUpdater(context, experiments))
+        `when`(experiments.getExperimentsUpdater(context)).thenReturn(experimentsUpdater)
+        experimentSource = source
+        `when`(experimentsUpdater.getExperimentSource(configuration)).thenReturn(experimentSource)
+
+        workRequest = experimentsUpdater.getWorkRequest()
+        `when`(experimentsUpdater.getWorkRequest()).thenReturn(workRequest)
+    }
+
+    @Before
+    fun setUp() {
+        resetExperiments()
+    }
+
+    @Test
+    fun `ExperimentsUpdater initialize schedules ExperimentsUpdateWorker in WorkManager`() {
+        assertFalse("ExperimentsUpdateWorker should not be scheduled yet",
+            isWorkScheduled(ExperimentsUpdater.TAG))
+        experimentsUpdater.initialize(configuration)
+        assertTrue("initialize must schedule ExperimentsUpdateWorker",
+            isWorkScheduled(ExperimentsUpdater.TAG))
+        val workInfo = getWorkInfoByTag(ExperimentsUpdater.TAG)
+        assertNotNull("workInfo must not be null", workInfo)
+        assertEquals("workInfo id must match request id", workRequest.id, workInfo?.id)
+    }
+
+    @Test
+    fun `ExperimentsUpdateWorker doWork returns SUCCESS`() {
+        experiments.initialize(context, configuration)
+        experimentsUpdater.initialize(configuration)
+        // Replace the updater in the singleton with the one built for the test
```

Comment:
Super nit: A few of the comments in this file (and possibly elsewhere?) are missing trailing ".".